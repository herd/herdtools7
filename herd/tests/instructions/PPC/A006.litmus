PPC A006

(* Test LWAX instruction*)
{ 0:r0 = 0; 0:r1 = x; uint32_t x = 0xffffffff;
  int64_t 0:r2;
  int64_t 0:r3; }
P0;
  lwax r2, r0, r1;
  lwzx r3, r0, r1;

forall 0:r2 = -1 /\ 0:r3 = 0xffffffff

