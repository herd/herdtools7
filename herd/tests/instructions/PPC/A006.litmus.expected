Test A006 Required
States 1
0:r2=-1; 0:r3=4294967295;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:r2=-1 /\ 0:r3=4294967295)
Observation A006 Always 1 0
Hash=110e6b64a73ded4540e9d15080165799

